client/components/LearnerViewDirective.js

```diff
@@ -410,11 +469,21 @@ tie.directive('learnerView', [function() {
           loadQuestion(questionId, questionSet.getIntroductionParagraphs());
         };
 
-        $scope.submitCode = function(code) {
+        var callSolutionHandlerService = function() {
+          var code = this.code;
           SolutionHandlerService.processSolutionAsync(
-            prompts[currentPromptIndex], code,
-            question.getAuxiliaryCode(language), language
-          ).then(setFeedback);
+              prompts[currentPromptIndex], code,
+              question.getAuxiliaryCode(language), language
+              ).then(setFeedback);
+        };
+
+        $scope.submitCode = function(code) {
+          $scope.hideDotDiv = false;
+          $timeout(
+              function(){feedbackDiv.scrollTop = feedbackDiv.scrollHeight + 17},
+              0);
+          var serviceCall = callSolutionHandlerService.bind({'code': code});
+          $timeout(serviceCall, 1000);
```

Comment:
What is the rationale for delaying the call for 1000ms? JS is single-threaded so the main point of the $timeout is just to force the angular $digest to happen (and the $scope values to be updated) before the code starts being processed. But I don't think we need an actual additional delay (at least from an infrastructure perspective), and 0 should work fine.